scribus/ui/fontcombo.cpp

```diff
@@ -320,3 +322,101 @@ void FontComboH::RebuildList(ScribusDoc *currentDoc, bool forAnnotation, bool fo
 	fontFamily->blockSignals(familySigBlocked);
 	fontStyle->blockSignals(styleSigBlocked);
 }
+
+fontFamilyDelegate::fontFamilyDelegate(QObject *parent):
+	QAbstractItemDelegate(parent)
+{
+}
+
+void fontFamilyDelegate::paint(QPainter *painter,
+							   const QStyleOptionViewItem &option,
+							   const QModelIndex &index) const
+{
+	QString text = index.data(Qt::DisplayRole).toString();
+	QFont font;
+	QStringList families = QFontDatabase().families();
+	QString name, style;
+	for (int i =0; i < families.size(); ++i)
```

Comment:
I don’t like the use of `QFontDatabase().families()` here. Scribus maintains its own font list and does not use this, so we should be using the real font list. But I don’t quite understand what the loop is for, why don’t you just call `QFontDatabase().font()` directly and check its result?